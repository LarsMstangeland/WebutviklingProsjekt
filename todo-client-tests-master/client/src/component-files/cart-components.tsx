import * as React from 'react';
import { Component } from 'react-simplified';
import { Alert, Card, Row, Column, Form, Button } from '../widgets';
import { createHashHistory } from 'history';
import cartService, {CartItem} from '../service-files/cart-service';
import { NavLink } from 'react-router-dom';
import recipeService from 'src/service-files/recipe-service';

const history = createHashHistory(); // Use history.push(...) to programmatically change path, for instance after successfully saving a student
//@ts-ignore
const userData = JSON.parse(sessionStorage.getItem('user'));

/**
 * Renders cart list.
 */
export class CartContent extends Component {
  //Array to store all cart
  cart:CartItem[] = [];
  CartItemsToShow:CartItem[] = [];
  searchBar: string = '';

  render() {
    return (
      <>
        <Card title="Your Cart">
          <Button.Danger onClick={() => {
            this.CartItemsToShow.map((cartitem) => {
              cartService.deleteIngredientFromCart(cartitem.cart_id)
            })
            this.mounted()
          }}>Clear All</Button.Danger>

          {this.CartItemsToShow.map((cart: CartItem) => (
            //Maps all the different cart and renders them as links to their respective cart details
            <Row key={cart.cart_id}>
              <Column>
                {cart.ingredients}
              </Column>
              <Column><Button.Danger onClick={() => {

                cartService.deleteIngredientFromCart(cart.cart_id).then(() => {
                  this.mounted();
                })
              }}>X</Button.Danger></Column>
            </Row>))}
        </Card>
      </>
    );
  }

  async mounted() {
    //Gets all cart and pass them to cart array
    try{
      let cart = await cartService.get(userData.user_id);
      this.cart = cart;
      this.CartItemsToShow = cart;

    } catch (error: any){
      Alert.danger('Error getting cart: ' + error.message)
    }
  }
}

