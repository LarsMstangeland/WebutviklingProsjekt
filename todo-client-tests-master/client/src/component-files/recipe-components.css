.react-simp {
  background-color: pink;
}

.picture-recipe {
  display: block;
  margin-left: auto;
  margin-right: auto;
  width: 50%;
}

/* The navbar container */
.topnav {
  overflow: hidden;
  background-color: #333;
}

/* Navbar links-------------mulig dette kan brukes */
.topnav a {
  float: left;
  display: block;
  color: #f2f2f2;
  text-align: center;
  padding: 14px 16px;
  text-decoration: none;
}

/* Links - change color on hover */
.topnav a:hover {
  background-color: #ddd;
  color: black;
}
