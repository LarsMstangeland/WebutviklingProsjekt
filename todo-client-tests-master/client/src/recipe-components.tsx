import * as React from 'react';
import { Component } from 'react-simplified';
import { Alert, Card, Row, Column, Form, Button } from './widgets';
import { NavLink } from 'react-router-dom';
import recipeService, { Recipe } from './recipe-service';
import { createHashHistory } from 'history';

const history = createHashHistory(); // Use history.push(...) to programmatically change path, for instance after successfully saving a student

/**
 * Renders recipe list.
 */
export class RecipeList extends Component {
  recipes: Recipe[] = [];

  render() {
    return (
      <>
        <Card title="Recipes">
          {this.recipes.map((recipe) => (
            <Row key={recipe.id}>
              <Column>
                <NavLink to={'/recipes/' + recipe.id}>{recipe.name}</NavLink>
              </Column>
            </Row>
          ))}
        </Card>
      </>
    );
  }

  mounted() {
    recipeService
      .getAll()
      .then((recipes) => (this.recipes = recipes))
      .catch((error) => Alert.danger('Error getting recipes: ' + error.message));
  }
}

/**
 * Renders a specific recipe.
 */
export class RecipeDetails extends Component<{ match: { params: { id: number } } }> {
  recipe: Recipe = { id: 0, name: '', description: '', region: '' };

  render() {
    return (
      <>
        <Card title="recipe">
          <Row>
            <Column width={2}>Title:</Column>
            <Column>{this.recipe.name}</Column>
          </Row>
          <Row>
            <Column width={2}>Description:</Column>
          </Row>
          <Row>
            <Column width={2}>Region:</Column>
          </Row>
        </Card>
      </>
    );
  }

  mounted() {
    recipeService
      .get(this.props.match.params.id)
      .then((recipe) => (this.recipe = recipe))
      .catch((error) => Alert.danger('Error getting recipe: ' + error.message));
  }
}