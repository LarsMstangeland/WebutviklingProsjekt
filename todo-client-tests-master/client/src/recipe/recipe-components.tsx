import * as React from 'react';
import { Component } from 'react-simplified';
import { Alert, Card, Row, Column, Form, Button } from '../widgets';
import { NavLink } from 'react-router-dom';
import recipeService, { Recipe, Ingredient } from './recipe-service';
import { createHashHistory } from 'history';

const history = createHashHistory(); // Use history.push(...) to programmatically change path, for instance after successfully saving a student

/**
 * Renders recipe list.
 */
export class RecipeList extends Component {
  recipes: Recipe[] = [];

  render() {
    return (
      <>
        <Card title="Recipes">
          {this.recipes.map((recipe) => (
            <Row key={recipe.recipe_id}>
              <Column>
                <NavLink to={'/recipes/' + recipe.recipe_id}>{recipe.name}</NavLink>
              
              </Column>
            </Row>))}
        </Card>
      </>
    );
  }

  mounted() {
    recipeService
      .getAll()
      .then((recipes) => (this.recipes = recipes))
      .catch((error) => Alert.danger('Error getting recipes: ' + error.message));
  }
}

/**
 * Renders a specific recipe.
 */
export class RecipeDetails extends Component<{ match: { params: { id: number } } }> {
  recipe: Recipe = { recipe_id: 0, name: '', description: '', region: '', picture_url: '' };
  ingredients: Ingredient[] = [];

  render() {
    
    return (
      <>
        {console.log(this.recipe)}
        <Card title={this.recipe.name}>
          <Row>
            <picture>
              <img src={this.recipe.picture_url} alt={this.recipe.name} />
            </picture>
          </Row>
          <Row>
            <Column width={2}>Region:</Column>
            <Column>{this.recipe.region}</Column>
          </Row>
          <Row>
            <Column width={2}>Description:</Column>
            <Column>{this.recipe.description}</Column>
          </Row>
          {this.ingredients.map((ingredient) => (
            <Row key={ingredient.ingredients_id}>
              <Column>{ingredient.name}</Column>
              <Column>{ingredient.amount}</Column>
              <Column>{ingredient.unit}</Column>
            </Row>
            ))}
          <Row>
            <Column><Button.Danger onClick={() => {
                recipeService.delete(this.recipe.recipe_id).then(() => {
                  history.push('/recipes');
            })
            }}>Delete</Button.Danger></Column>
            <Column><Button.Success onClick={() => {
              history.push('/recipe/' + this.props.match.params.id + '/edit')
            }}>Edit</Button.Success></Column>
          </Row>
        </Card>
      </>
    );
  }

  async mounted() {
    /*
    recipeService
      .get(this.props.match.params.id)
      .then((recipe) => (this.recipe = recipe)).then(()=>{
        recipeService
        .getRecipeIngredients(this.recipe.recipe_id)
        .then((ingredients) => ( this.ingredients = ingredients)).then(()=>{console.log(this.ingredients)})
        .catch((error) => Alert.danger('Error getting ingredients: ' + error.message));
      })
      .catch((error) => Alert.danger('Error getting recipe: ' + error.message));
      */
     try {
      let recipe = await recipeService.get(this.props.match.params.id)
      this.recipe = recipe;
      let ingredients = await recipeService.getRecipeIngredients(this.props.match.params.id)
      this.ingredients = ingredients;

     } catch (error: any) {
      Alert.danger('Error getting recipe or ingredients: ' + error.message)
     }
      
  }
}

export class RecipeEdit extends Component<{ match: { params: { id: number } } }> {
  recipe: Recipe = { recipe_id: 0, name: '', description: '', region: '', picture_url: '' };
  ingredients: Ingredient[] = [];

  render() {
    return (
      <>
        <Card title="Edit task">
          <Row>
            <Column width={2}>
              <Form.Label>Name:</Form.Label>
            </Column>
            <Column>
              <Form.Input
                type="text"
                value={this.recipe.name}
                onChange={(event) => (this.recipe.name = event.currentTarget.value)}
              />
            </Column>
          </Row>
          <Row>
            <Column width={2}>
              <Form.Label>Description:</Form.Label>
            </Column>
            <Column>
              <Form.Textarea
                value={this.task.description}
                onChange={(event) => {
                  this.task.description = event.currentTarget.value;
                }}
                rows={10}
              />
            </Column>
          </Row>
          <Row>
            <Column width={2}>Done:</Column>
            <Column>
              <Form.Checkbox
                checked={this.task.done}
                onChange={(event) => (this.task.done = event.currentTarget.checked)}
              />
            </Column>
          </Row>
        </Card>
        <Row>
          <Column>
            <Button.Success
              onClick={() =>
                taskService.update(this.task).then(() => {
                  history.push('/tasks/' + this.task.id);
                })
              }
            >
              Save
            </Button.Success>
          </Column>
          <Column right>
            <Button.Danger
              onClick={() =>
                taskService.delete(this.task.id).then(() => {
                  history.push('/tasks');
                })
              }
            >
              Delete
            </Button.Danger>
          </Column>
        </Row>
      </>
    );
  }

  mounted() {
    taskService
      .get(this.props.match.params.id)
      .then((task) => (this.task = task))
      .catch((error) => Alert.danger('Error getting task: ' + error.message));
  }
}