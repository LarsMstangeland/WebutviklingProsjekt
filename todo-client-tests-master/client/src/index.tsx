import ReactDOM from 'react-dom';
import * as React from 'react';
import { Component } from 'react-simplified';
import { HashRouter, Route } from 'react-router-dom';
import { NavBar, Card, Alert, PreviewCard } from './widgets';
import { RecipeList, RecipeDetails, RecipeEdit } from './component-files/recipe-components';
import {UserLogin} from './component-files/user-components';

class Menu extends Component {
  render() {
    return (
      <NavBar brand="Food Junkies">
        <NavBar.Link to="/recipes">Recipes</NavBar.Link>
        <NavBar.Link to="/my_profile">My Profile</NavBar.Link>
      </NavBar>
    );
  }
}

class Home extends Component {
  render() {
    return (
      <div>
        <Card title="Welcome">To Food Junkies</Card>

          <PreviewCard title1 ="dette er en tittel1" title2 = "dette er en tittel2" image1 = "https://i.guim.co.uk/img/media/26392d05302e02f7bf4eb143bb84c8097d09144b/446_167_3683_2210/master/3683.jpg?width=1200&quality=85&auto=format&fit=max&s=a52bbe202f57ac0f5ff7f47166906403" image2 = "https://i.guim.co.uk/img/media/26392d05302e02f7bf4eb143bb84c8097d09144b/446_167_3683_2210/master/3683.jpg?width=1200&quality=85&auto=format&fit=max&s=a52bbe202f57ac0f5ff7f47166906403">
          </PreviewCard>


      </div>

    )

  }
}

ReactDOM.render(
  <HashRouter>
    <div>
      <Alert />
      <Menu />
      <Route exact path="/" component={Home} />
      <Route exact path="/recipes" component={RecipeList} />
      <Route exact path="/recipes/:id(\d+)" component={RecipeDetails} /> {/* id must be number */}
      <Route exact path='/recipes/:id(\d+)/edit' component={RecipeEdit} />
      <Route exact path='/my_profile' component={UserLogin}/>
    </div>
  </HashRouter>,
  document.getElementById('root')
);
