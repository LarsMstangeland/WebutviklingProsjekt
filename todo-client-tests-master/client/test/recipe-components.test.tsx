import * as React from 'react';
import { RecipeList, RecipeDetails, RecipeEdit } from '../src/component-files/recipe-components';
import { shallow } from 'enzyme';
import { Form, Button, PreviewCard, Column, Alert } from '../src/widgets';
import { NavLink } from 'react-router-dom';

jest.mock('../src/service-files/recipe-service', () => {
  class RecipeService {
    get() {
      return Promise.resolve({
        recipe_id: 2,
        name: 'Hamburger',
        region: 'Asia',
        picture_url: '',
        description: 'Digg',
        type: 'Beef',
      });
    }

    getAll() {
      return Promise.resolve([
        {
          recipe_id: 1,
          name: 'Hotdog',
          region: 'Europe',
          picture_url: '',
          description: 'Namnam',
          type: 'Beef',
        },
        {
          recipe_id: 2,
          name: 'Hamburger',
          region: 'Asia',
          picture_url: '',
          description: 'Digg',
          type: 'Beef',
        },
        {
          recipe_id: 3,
          name: 'Pizza',
          region: 'Asia',
          picture_url: '',
          description: 'Mhmmm',
          type: 'Vegatarian',
        },
      ]);
    }

    getRecipeIngredients() {
      return Promise.resolve([
        { ingredients_id: 1, name: 'water', amount: '1', unit: 'dl' },
        { ingredients_id: 2, name: 'chicken', amount: '200', unit: 'g' },
      ]);
    }

    getAllIngredients() {
      return Promise.resolve([
        { ingredients_id: 1, name: 'water' },
        { ingredients_id: 2, name: 'chicken' },
        { ingredients_id: 3, name: 'chili' },
        { ingredients_id: 4, name: 'pepper' },
      ]);
    }
  }
  return new RecipeService();
});

jest.mock('../src/service-files/utility-service.tsx', () => {
  class UtilityService {
    getAllRegions() {
      return Promise.resolve([
        { id: 1, name: 'Europe' },
        { id: 2, name: 'Asia' },
      ]);
    }

    getAllTypes() {
      return Promise.resolve([
        { id: 1, name: 'Beef' },
        { id: 2, name: 'Pork' },
        { id: 3, name: 'Vegatarian' },
      ]);
    }

    getAllUnits() {
      return Promise.resolve([
        { id: 1, unit: 'dl' },
        { id: 2, unit: 'g' },
      ]);
    }
  }
  return new UtilityService();
});

describe('RecipeList tests', () => {
  test('RecipeList draws correctly - snapshot', () => {
    const wrapper = shallow(<RecipeList />);

    expect(wrapper).toMatchSnapshot();
  });

  test('Region input filters correct', (done) => {
    const wrapper = shallow(<RecipeList />);

    setTimeout(() => {
      expect(
        wrapper.containsAllMatchingElements([
          <PreviewCard id={1} url="" name="Hotdog"></PreviewCard>,
          <PreviewCard id={2} url="" name="Hamburger"></PreviewCard>,
          <PreviewCard id={3} url="" name="Pizza"></PreviewCard>,
        ])
      ).toEqual(true);
    });

    wrapper
      .find(Form.Select)
      .at(0)
      .simulate('change', { currentTarget: { value: 'Asia' } });

    setTimeout(() => {
      expect(
        wrapper.containsAllMatchingElements([
          <PreviewCard id={2} url="" name="Hamburger"></PreviewCard>,
          <PreviewCard id={3} url="" name="Pizza"></PreviewCard>,
        ])
      ).toEqual(true);
    });
    done();
  });

  test('Type input filters correct', (done) => {
    const wrapper = shallow(<RecipeList />);

    setTimeout(() => {
      expect(
        wrapper.containsAllMatchingElements([
          <PreviewCard id={1} url="" name="Hotdog"></PreviewCard>,
          <PreviewCard id={2} url="" name="Hamburger"></PreviewCard>,
          <PreviewCard id={3} url="" name="Pizza"></PreviewCard>,
        ])
      ).toEqual(true);
    });

    wrapper
      .find(Form.Select)
      .at(1)
      .simulate('change', { currentTarget: { value: 'Beef' } });

    setTimeout(() => {
      expect(
        wrapper.containsAllMatchingElements([
          <PreviewCard id={1} url="" name="Hotdog"></PreviewCard>,
          <PreviewCard id={2} url="" name="Hamburger"></PreviewCard>,
        ])
      ).toEqual(true);
    });
    done();
  });

  test('Test to see if recipe in list push correct location', (done) => {
    const wrapper = shallow(<RecipeList />);

    setTimeout(() => {
      wrapper.find(PreviewCard).at(0).simulate('click');
    });

    setTimeout(() => {
      expect(location.hash).toEqual('#/recipes/1');
    });

    done();
  });
});

describe('RecipeDetails tests', () => {
  test('RecipeDetails draws correctly - snapshot', () => {
    const wrapper = shallow(<RecipeDetails match={{ params: { id: 1 } }} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('RecipeDetails draws correctly when not logged in', (done) => {
    const wrapper = shallow(<RecipeDetails match={{ params: { id: 2 } }} />);

    setTimeout(() => {
      expect(
        wrapper.containsAllMatchingElements([
          <h2>Hamburger</h2>,
          <div>
            <p>Region:</p>
            <p>Asia</p>
          </div>,
          <div>
            <p>Type:</p>
            <p>Beef</p>
          </div>,
          <div>
            <p>Description:</p>
            <p>Digg</p>
          </div>,
          <PreviewCard id={1} url="" name="Hotdog"></PreviewCard>,
          <PreviewCard id={3} url="" name="Pizza"></PreviewCard>,
        ])
      ).toEqual(true);
    });

    wrapper.find(Button.Success).at(0).simulate('click');

    expect(
      wrapper.containsMatchingElement(
        <div>
          Log in to like a recipe
          <button />
        </div>
      )
    );

    done();
  });

  test('RecipeDetails draws correctly when logged in', (done) => {
    const wrapper = shallow(<RecipeDetails match={{ params: { id: 2 } }} />);

    const userData = { user_id: 1, username: 'larsy', password: 'larserkul', admin: true };
    window.sessionStorage.setItem('user', JSON.stringify(userData));

    wrapper.find(Button.Success).at(0).simulate('click');

    setTimeout(() => {
      expect(
        wrapper.containsAllMatchingElements(
          //@ts-ignore
          <Button.Danger>Unlike</Button.Danger>,
          //@ts-ignore
          <Button.Danger>Delete</Button.Danger>,
          //@ts-ignore
          <Button.Success>Edit</Button.Success>
        )
      );
    });

    done();
  });

  test('Sets location to edit page on click', (done) => {
    const wrapper = shallow(<RecipeDetails match={{ params: { id: 2 } }} />);

    wrapper.find(Button.Success).at(2).simulate('click');

    setTimeout(() => {
      expect(location.hash).toEqual('#/recipes/2/edit');
    });

    done();
  });

  test('Sets location to recipe list page on click', (done) => {
    const wrapper = shallow(<RecipeDetails match={{ params: { id: 2 } }} />);

    setTimeout(() => {
      wrapper.find(Button.Danger).at(0).simulate('click');
    });

    setTimeout(() => {
      expect(location.hash).toEqual('#/recipes');
    });

    done();
  });

  test('Test if amount in ingredients is responsive', () => {
    const wrapper = shallow(<RecipeDetails match={{ params: { id: 2 } }} />);

    setTimeout(() => {
      expect(wrapper.containsAllMatchingElements([<Column>1</Column>, <Column>200</Column>]));
    });

    wrapper.find(Form.Input).simulate('change', { currentTarget: { value: 2 } });

    setTimeout(() => {
      expect(wrapper.containsAllMatchingElements([<Column>0.5</Column>, <Column>100</Column>]));
    });
  });

  test('Test like button', (done) => {
    const wrapper = shallow(<RecipeDetails match={{ params: { id: 2 } }} />);

    setTimeout(() => {
      wrapper.find(Button.Success).at(0).simulate('click');
    });

    setTimeout(() => {
      expect(
        //@ts-ignore
        wrapper.containsMatchingElement(<Button.Danger>Unlike</Button.Danger>)
      );
    });

    setTimeout(() => {
      wrapper.find(Button.Danger).at(0).simulate('click');
    });

    setTimeout(() => {
      expect(
        //@ts-ignore
        wrapper.containsMatchingElement(<Button.Success>Like</Button.Success>)
      );
    });

    done();
  });
});

describe('RecipeEdit tests', () => {
  test('RecipeEdit draws correctly - snapshot', () => {
    const wrapper = shallow(<RecipeEdit match={{ params: { id: 1 } }} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('RecipeEdit draws correctly', (done) => {
    const wrapper = shallow(<RecipeEdit match={{ params: { id: 2 } }} />);

    setTimeout(() => {
      expect(
        wrapper.containsAllMatchingElements([
          <h2>Hamburger</h2>,
          //@ts-ignore
          <Form.Input value="Hamburger"></Form.Input>,
          //@ts-ignore
          <Form.Select value="Asia"></Form.Select>,
          //@ts-ignore
          <Form.Select value="Beef"></Form.Select>,
          //@ts-ignore
          <Form.Textarea value="Digg"></Form.Textarea>,
          <Column>Water</Column>,
          //@ts-ignore
          <Form.Input value="1"></Form.Input>,
          //@ts-ignore
          <Form.Select value="dl"></Form.Select>,
          <Column>Chicken</Column>,
          //@ts-ignore
          <Form.Input value="200"></Form.Input>,
          //@ts-ignore
          <Form.Select value="g"></Form.Select>,
        ])
      ).toEqual(true);
    });

    done();
  });

  test('RecipeEdit add ingredient to recipe', (done) => {
    const wrapper = shallow(<RecipeEdit match={{ params: { id: 2 } }} />);

    setTimeout(() => {
      expect(
        wrapper.containsAllMatchingElements([
          <Column>Water</Column>,
          //@ts-ignore
          <Form.Input value="1"></Form.Input>,
          //@ts-ignore
          <Form.Select value="dl"></Form.Select>,
          <Column>Chicken</Column>,
          //@ts-ignore
          <Form.Input value="200"></Form.Input>,
          //@ts-ignore
          <Form.Select value="g"></Form.Select>,
        ])
      ).toEqual(true);
    });

    wrapper
      .find(Form.Input)
      .at(2)
      .simulate('change', { currentTarget: { value: 'chi' } });

    setTimeout(() => {
      expect(
        wrapper.containsMatchingElement(
          <datalist>
            <option value={'chicken'}>chicken</option>
            <option value={'chili'}>chili</option>
          </datalist>
        )
      ).toEqual(true);
    });

    wrapper
      .find(Form.Input)
      .at(2)
      .simulate('change', { currentTarget: { value: 'rice' } });
    wrapper
      .find(Form.Input)
      .at(3)
      .simulate('change', { currentTarget: { value: '100' } });
    wrapper
      .find(Form.Select)
      .at(2)
      .simulate('change', { currentTarget: { value: 'g' } });
    wrapper.find(Button.Light).at(0).simulate('click');

    setTimeout(() => {
      expect(
        wrapper.containsAllMatchingElements([
          <Column>Water</Column>,
          //@ts-ignore
          <Form.Input value="1"></Form.Input>,
          //@ts-ignore
          <Form.Select value="dl"></Form.Select>,
          <Column>Chicken</Column>,
          //@ts-ignore
          <Form.Input value="200"></Form.Input>,
          //@ts-ignore
          <Form.Select value="g"></Form.Select>,
          <Column>Rice</Column>,
          //@ts-ignore
          <Form.Input value="100"></Form.Input>,
          //@ts-ignore
          <Form.Select value="g"></Form.Select>,
        ])
      ).toEqual(true);
    });

    setTimeout(() => {
      expect(
        wrapper.containsAllMatchingElements(
          //@ts-ignore
          <Button.Danger>Unlike</Button.Danger>,
          //@ts-ignore
          <Button.Danger>Delete</Button.Danger>,
          //@ts-ignore
          <Button.Success>Edit</Button.Success>
        )
      );
    });

    done();
  });

  test('RecipeEdit sets correct location after save', (done) => {
    const wrapper = shallow(<RecipeEdit match={{ params: { id: 2 } }} />);

    wrapper.find(Button.Success).simulate('click');

    setTimeout(() => {
      expect(location.hash).toEqual('#/recipes/2');
    });

    done();
  });

  test('Delete and add ingredient to recipe', (done) => {
    const wrapper = shallow(<RecipeEdit match={{ params: { id: 2 } }} />);

    setTimeout(() => {
      wrapper.find(Button.Danger).at(0).simulate('click');
    });

    setTimeout(() => {
      expect(
        //@ts-ignore
        wrapper.containsMatchingElement(<Button.Success>Add</Button.Success>)
      );
    });

    wrapper.find(Button.Success).simulate('click');

    setTimeout(() => {
      expect(
        wrapper.containsAllMatchingElements(
          //@ts-ignore
          <Button.Danger>X</Button.Danger>,
          //@ts-ignore
          <Button.Danger>X</Button.Danger>
        )
      );
    });

    done();
  });
});
