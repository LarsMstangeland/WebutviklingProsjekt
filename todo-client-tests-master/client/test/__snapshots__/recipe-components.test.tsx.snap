// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeDetails tests RecipeDetails draws correctly - snapshot 1`] = `
<Fragment>
  <div
    style={
      {
        "marginLeft": "5rem",
        "marginTop": "2rem",
        "padding": "1rem",
      }
    }
  >
    <div
      style={
        {
          "display": "flex",
          "marginBottom": "2rem",
        }
      }
    >
      <img
        alt=""
        src=""
        style={
          {
            "borderRadius": "2rem",
            "height": "auto",
            "width": "50%",
          }
        }
      />
      <div
        style={
          {
            "display": "flex",
            "flexDirection": "column",
            "marginLeft": "2rem",
            "width": "50%",
          }
        }
      >
        <h2
          style={
            {
              "marginBottom": "1rem",
            }
          }
        />
        <bound RecipeDetail
          name="Region"
          value=""
        />
        <bound RecipeDetail
          name="Type"
          value=""
        />
        <bound RecipeDetail
          name="Description"
          value=""
        />
      </div>
    </div>
    <div
      style={
        {
          "display": "flex",
          "justifyContent": "space-around",
          "marginBottom": "0.5rem",
          "width": "27.5vw",
        }
      }
    >
      <ButtonSuccess
        onClick={[Function]}
      >
        Like recipe
      </ButtonSuccess>
      <ButtonSuccess
        onClick={[Function]}
      >
        Add ingredients to cart
      </ButtonSuccess>
      <ButtonLight
        onClick={[Function]}
      >
        Share
      </ButtonLight>
    </div>
    <div>
      <br />
      <h4>
        Ingredients
      </h4>
      <Row>
        <Column>
          Portions:
        </Column>
        <Column>
          <FormInput
            max="50"
            min="1"
            onChange={[Function]}
            type="number"
            value={4}
          />
        </Column>
      </Row>
      <br />
      <Row>
        <Column>
          <b>
            Ingredients name:
          </b>
        </Column>
        <Column>
          <b>
            Amount:
          </b>
        </Column>
        <Column>
          <b>
            Unit:
          </b>
        </Column>
      </Row>
      <br />
       
    </div>
    <br />
  </div>
</Fragment>
`;

exports[`RecipeEdit tests RecipeEdit draws correctly - snapshot 1`] = `
<Fragment>
  <Card
    title="Edit recipe"
  >
    <Row
      style={
        {
          "marginBottom": "0.5rem",
        }
      }
    >
      <Column
        width={2}
      >
        <FormLabel>
          Name:
        </FormLabel>
      </Column>
      <Column>
        <FormInput
          onChange={[Function]}
          type="text"
          value=""
        />
      </Column>
    </Row>
    <Row>
      <Column
        width={2}
      >
        <FormLabel>
          Region:
        </FormLabel>
      </Column>
      <Column>
        <FormSelect
          onChange={[Function]}
          value=""
        >
          <option
            value="Regions"
          >
            Regions
          </option>
        </FormSelect>
      </Column>
    </Row>
    <Row>
      <Column
        width={2}
      >
        <FormLabel>
          Type:
        </FormLabel>
      </Column>
      <Column>
        <FormSelect
          onChange={[Function]}
          value=""
        >
          <option
            value="Types"
          >
            Types
          </option>
        </FormSelect>
      </Column>
    </Row>
    <Row
      style={
        {
          "marginBottom": "0.5rem",
        }
      }
    >
      <Column
        width={2}
      >
        <FormLabel>
          Description:
        </FormLabel>
      </Column>
      <Column>
        <FormTextarea
          onChange={[Function]}
          rows={10}
          value=""
        />
      </Column>
    </Row>
    <Row>
      <Column
        width={2}
      >
        Image URL:
      </Column>
      <Column>
        <FormInput
          onChange={[Function]}
          type="text"
          value=""
        />
      </Column>
    </Row>
    <br />
    <Row>
      <Column>
        <b>
          Ingredients name:
        </b>
      </Column>
      <Column>
        <b>
          Amount:
        </b>
      </Column>
      <Column>
        <b>
          Unit:
        </b>
      </Column>
      <Column />
    </Row>
    <br />
    <h4>
      Add ingredient
    </h4>
    <Row>
      <Column>
        <FormInput
          list="searchList"
          onChange={[Function]}
          placeholder="Search for ingredient"
          type="search"
          value=""
        />
        <datalist
          id="searchList"
        />
      </Column>
      <Column>
        <FormInput
          max="1000"
          onChange={[Function]}
          type="number"
          value={0}
        />
      </Column>
      <Column>
        <FormSelect
          onChange={[Function]}
          value=""
        >
          <option>
            Select Unit
          </option>
        </FormSelect>
      </Column>
      <Column>
        <ButtonLight
          onClick={[Function]}
          small={true}
        >
          Add
        </ButtonLight>
      </Column>
    </Row>
  </Card>
  <Row>
    <Column>
      <ButtonSuccess
        onClick={[Function]}
      >
        Save
      </ButtonSuccess>
    </Column>
  </Row>
</Fragment>
`;

exports[`RecipeList tests RecipeList draws correctly - snapshot 1`] = `
<div>
  <div
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "margin": "1rem",
        "padding": "1rem",
      }
    }
  >
    <h1>
      Recipes
    </h1>
    <br />
    <div
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
          "marginBottom": "1rem",
          "width": "100%",
        }
      }
    >
      <div
        style={
          {
            "margin": "0 0.5rem",
          }
        }
      >
        <FormInput
          onChange={[Function]}
          placeholder="Search for recipes"
          style={
            {
              "border": "2px solid black",
              "height": "5.5vh",
              "width": "12vw",
            }
          }
          type="search"
          value=""
        />
      </div>
      <div
        style={
          {
            "margin": "0 0.5rem",
          }
        }
      >
        <FormSelect
          onChange={[Function]}
          style={
            {
              "border": "2px solid black",
              "borderRadius": "5px",
              "height": "5.5vh",
              "width": "12vw",
            }
          }
          value="Region"
        >
          <option>
            Region
          </option>
        </FormSelect>
      </div>
      <div
        style={
          {
            "margin": "0 0.5rem",
          }
        }
      >
        <FormSelect
          onChange={[Function]}
          style={
            {
              "border": "2px solid black",
              "borderRadius": "5px",
              "height": "5.5vh",
              "width": "12vw",
            }
          }
          value="Type"
        >
          <option>
            Type
          </option>
        </FormSelect>
      </div>
    </div>
    <div
      style={
        {
          "display": "flex",
          "flexWrap": "wrap",
          "justifyContent": "center",
        }
      }
    >
      <h3>
        No results
      </h3>
    </div>
  </div>
</div>
`;
